modules/investigating-worker-node-installation-issues.adoc

```diff
@@ -0,0 +1,200 @@
+----
+
+. Monitor `kubelet.service` journald unit logs on worker nodes, after they have booted. This provides visibility into worker node agent activity.
+.. Retrieve the logs using `oc`:
++
+----
+$ oc adm node-logs --role=worker -u kubelet
+----
++
+.. If the API is not functional, review the logs using SSH instead. Replace `<worker-node>.<cluster_name>.<base_domain>` with appropriate values:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> journalctl -b -f -u kubelet.service
+----
++
+[NOTE]
+====
+{product-title} {product-version} cluster nodes running {op-system-first} are immutable and rely on Operators to apply cluster changes. Accessing cluster nodes using SSH is not recommended and nodes will be tainted as _accessed_. Before attempting to collect diagnostic data over SSH, review whether the data collected by running `oc adm must gather` and other `oc` commands is sufficient instead. However, if the {product-title} API is not available, or the kubelet is not properly functioning on the target node, `oc` operations will be impacted. In such situations, it is possible to access nodes using `ssh core@<node>.<cluster_name>.<base_domain>`.
+====
++
+. Retrieve `crio.service` journald unit logs on worker nodes, after they have booted. This provides visibility into worker node CRI-O container runtime activity.
+.. Retrieve the logs using `oc`:
++
+----
+$ oc adm node-logs --role=worker -u crio
+----
++
+.. If the API is not functional, review the logs using SSH instead:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> journalctl -b -f -u crio.service
+----
+
+. Collect logs from specific subdirectories under `/var/log/` on worker nodes.
+.. Retrieve a list of logs contained within a `/var/log/` subdirectory. The following example lists files in `/var/log/sssd/` on all worker nodes:
++
+----
+$ oc adm node-logs --role=worker --path=sssd
+----
++
+.. Inspect a specific log within a `/var/log/` subdirectory. The following example outputs `/var/log/sssd/audit.log` contents from all worker nodes:
++
+----
+$ oc adm node-logs --role=worker --path=sssd/sssd.log
+----
++
+.. If the API is not functional, review the logs on each node using SSH instead. The following example tails `/var/log/sssd/sssd.log`:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> sudo tail -f /var/log/sssd/sssd.log
+----
+
+. Review worker node container logs using SSH.
+.. List the containers:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> sudo crictl ps -a
+----
++
+.. Retrieve a container's logs using `crictl`:
++
+----
+$ ssh core@<worker-node>.<cluster_name>.<base_domain> sudo crictl logs -f <container_id>
+----
+
+. If you experience worker node configuration issues, verify that the MCO, MCO endpoint and DNS record are functioning. The Machine Config Operator (MCO) manages operating system configuration during the installation procedure. Also verify system clock accuracy and certificate validity.
```

Comment:
s/MCO, MCO endpoint and DNS/MCO, MCO endpoint, and DNS/

Reply:
Resolved.